From 00530eabab11d89bc586fbf88dd70a3d6a8d0e66 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Thu, 7 Aug 2014 07:06:32 -0400
Subject: [net] vti4: Enable namespace changing

Message-id: <1407395194-3809-17-git-send-email-jpirko@redhat.com>
Patchwork-id: 87530
O-Subject: [patch rhel7 net 16/18] vti4: Enable namespace changing
Bugzilla: 1091561
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

BZ1091561
https://brewweb.devel.redhat.com/taskinfo?taskID=7794261
Tested by jtluka's LNST smoke test recipe

upstream commit 895de9a3488abcdd186680f0af3cce7f2d4d4a6e
Author: Steffen Klassert <steffen.klassert@secunet.com>
Date:   Fri Feb 21 08:41:11 2014 +0100

    vti4: Enable namespace changing

    vti4 is now fully namespace aware, so allow namespace changing
    for vti devices

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/ip_vti.c b/net/ipv4/ip_vti.c
index 21f1624..67a19b3 100644
--- a/net/ipv4/ip_vti.c
+++ b/net/ipv4/ip_vti.c
@@ -354,7 +354,6 @@ static int vti_tunnel_init(struct net_device *dev)
  dev->flags  = IFF_NOARP;
  dev->iflink  = 0;
  dev->addr_len  = 4;
- dev->features  |= NETIF_F_NETNS_LOCAL;
  dev->features  |= NETIF_F_LLTX;
  dev->priv_flags  &= ~IFF_XMIT_DST_RELEASE;
 
-- 
1.7.1